pg_search/src/postgres/merge.rs

```diff
@@ -314,7 +314,7 @@ unsafe extern "C-unwind" fn background_merge(arg: pg_sys::Datum) {
         );
 
         let current_xid = pg_sys::GetCurrentFullTransactionId();
-        let next_xid = current_xid;
+        let next_xid = pg_sys::ReadNextFullTransactionId();
```

Comment:
AFAICT, a background worker runs in its own transaction: https://github.com/paradedb/paradedb/blob/9a042d016bee581a25527035f34fb99c93330e78/pg_search/src/postgres/merge.rs#L306-L307 ... which means that its transaction ID never changes during all of the merges that it does, until it commits.

That should mean that it is the only one which is able to consume blocks freed by the merges that it does, right? At least until after it has committed.